lib/galaxy/datatypes/assembly.py

```diff
@@ -77,8 +77,7 @@ def sniff_prefix(self, file_prefix):
           CGACGAATGACAGGTCACGAATTTGGCGGGGATTA
         """
         fh = file_prefix.string_io()
-        while True:
-            line = fh.readline()
+        for line in fh.readlines():
             if not line:
                 break  # EOF
```

Comment:
No need to use `readlines()`, which would load the `fh` StringIO in memory as a list.

```suggestion
        for line in fh:
```